### PR TITLE
 Changes in Integer Partition

### LINKED ISSUE #10115: Adding Changes in Integer Partition
### Feature description

The code calculates the number of ways to partition a positive integer m into distinct positive integers and provides user input validation and command-line argument handling.

### PR DESCRIPTION
**Pull request by imSanko**
### Describe your change:

Partition a positive integer m into distinct positive integers and provides user input validation and command-line argument handling.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #10115".

### CODE REVIEW

**Comment by imSanko** [comment]
@cclauss Please review this pull request. It has passed all necessary checks. Confirm if we can proceed with moving forward. Thank you for your time and consideration.

**Comment by imSanko** [comment]
can you check the PR cause, I've added the changes and the check have been passed kindly @cclauss have a look.

**Comment by imSanko** [comment]
@cclauss can you please check the pr and let me know.

**Comment by imSanko** [comment]
@tianyizheng02 can you check

File: sorts/binary_insertion_sort.py [hunk 1]
```diff
@@ -54,8 +33,50 @@ def binary_insertion_sort(collection: list) -> list:
         collection[low] = val
     return collection
 
+def test_binary_insertion_sort():
```
Comment (cclauss) [inline sorts/binary_insertion_sort.py:39]: Are these tests being run in our `build` GitHub Action?  I think not.

Comment (imSanko) [reply sorts/binary_insertion_sort.py:39]: Yes, It does ran and passed all of them @cclauss

**Pull request closed without merging**